### PR TITLE
[NTWK-562] Add test for packet capture info

### PR DESCRIPTION
**Pull request by pimlu**
### What does this PR do?
Adds tests to #30396 to make sure it doesn't panic, and tests the pktType logic
### Describe how to test/QA your changes

`sudo -E go test -tags=linux,linux_bpf,npm,process,test ./pkg/network/filter -v`

### CODE REVIEW

File: pkg/network/filter/packet_source_linux.go [hunk 1]
```diff
@@ -129,8 +129,15 @@ func (p *AFPacketSource) SetBPF(filter []bpf.RawInstruction) error {
 	return p.TPacket.SetBPF(filter)
 }
 
-// VisitPackets starts reading packets from the source
-func (p *AFPacketSource) VisitPackets(exit <-chan struct{}, visit func(data []byte, info PacketInfo, t time.Time) error) error {
+type zeroCopyPacketReader interface {
```
Comment (hmahmood) [inline pkg/network/filter/packet_source_linux.go:132]: :+1:

File: pkg/network/filter/packet_source_linux_test.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		CaptureLength:  0,
+		Length:         0,
+		InterfaceIndex: 0,
+		AncillaryData:  ancillaryData,
+	}
+}
+
+func expectAncillaryPktType(t *testing.T, ancillaryData []interface{}, pktType uint8) {
+	exit := make(chan struct{}, 1)
+	exit <- struct{}{}
```
[the remark this answers is not in the document]
Comment (pimlu) [reply pkg/network/filter/packet_source_linux_test.go:41]: Oh whoops, I need to remove that. I was troubleshooting the channel earlier

[the remark this answers is not in the document]
Comment (pimlu) [reply pkg/network/filter/packet_source_linux_test.go]: Updated, thanks for catching

**Review by hmahmood — APPROVED** [review]

**Pull request merged**